test/src/test/java/se/fortnox/reactivewizard/test/WaitingTestUtilsTest.java

```diff
@@ -0,0 +1,77 @@
+package se.fortnox.reactivewizard.test;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class WaitingTestUtilsTest {
+
+    @Test
+    public void shouldFailTestIfConditionIsNotMetWithinDefaultTime() {
+        Error assertionError = null;
+        try {
+            WaitingTestUtils.assertConditionIsTrueWithin5Seconds(() -> false);
+        } catch (Error error) {
+            assertionError = error;
+        }
+
+        assertThat(assertionError).isNotNull();
+    }
+
+    @Test
+    public void shouldFailTestIfConditionIsNotMetWithinDefaultTimeWithCustomMessage() {
+        Error assertionError = null;
+        String errorMessage = UUID.randomUUID().toString();
+        try {
+            WaitingTestUtils.assertConditionIsTrueWithin5Seconds(() -> false, errorMessage);
+        } catch (AssertionError error) {
+            assertionError = error;
+        }
+
+        assertThat(assertionError).isNotNull();
+        assertThat(assertionError.getMessage()).isEqualTo(errorMessage);
+    }
+
+    @Test
+    public void shouldFailTestIfConditionIsNotMetWithinTime() {
+        Error assertionError = null;
+        try {
+            WaitingTestUtils.assertConditionIsTrueWithinTime(3000, TimeUnit.MILLISECONDS, () -> false);
```

Comment:
Nitpick: adding 3 seconds for a test feels like a bit much, the functionallity should be able to be tested with 50ms?

Reply:
The testutil checks every 100ms so I will raise it to the least possible value